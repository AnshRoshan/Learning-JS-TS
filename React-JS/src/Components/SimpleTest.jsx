function SimpleTest({ username, email, age }) {
	return (
		<div>
			<h1>Simple Test</h1>
			<p>This is a simple test component</p>
			<div>
				<p>Username: {username}</p>
				<p>Email: {email}</p>
				<p>Age: {age}</p>
			</div>
		</div>
	)
}
export default SimpleTest
