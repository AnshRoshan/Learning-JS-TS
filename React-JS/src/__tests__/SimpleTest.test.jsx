import { render, screen } from '@testing-library/react'
import { describe, expect, it } from 'vitest'
import SimpleTest from '../components/SimpleTest'

describe('Check if the Vitest is working', () => {
	it('should return true', () => {
		expect(true).toBe(true)
	})

	it('GetByTest the element in the component', () => {
		render(<SimpleTest username={'@ansh'} age={22} email={'anshroshan@gmail.com'} />)
		// screen.debug()
		const elements = screen.getAllByText(/ansh/)
		elements.forEach((element) => {
			expect(element).toBeInTheDocument()
		})
	})

	it('QueryByTest the element in the component', () => {
		render(<SimpleTest username={'@ansh'} age={22} email={'anshroshan@gmail.com'} />)
		const elements = screen.queryAllByText(/ansh78/)
		// expect(elements.length).toBeGreaterThan(1)
		elements.forEach((element) => {
			expect(element).toBeInTheDocument()
		})
	})

	// using the query to check if the elwmwnt is not present in the component
	it('QueryByTest the element not to be present in the component', () => {
		render(<SimpleTest username={'@ansh'} age={22} email={'anshroshan@gmail.com'} />)
		expect(screen.queryByText(/ansh78/)).not.toBeInTheDocument()
		// const elements = screen.queryAllByText(/ansh78/)
		// expect(elements.length).toBe(0)
	})

	it('FindBy Test the element in the component', async () => {
		render(<SimpleTest username={'@ansh'} age={22} email={'ansh@test.com'} />)
		expect(await screen.findByText(/ansh@test/, {}, { timeout: 7000 })).toBeInTheDocument()
	})

	it('Snapshot testing : Simpletest', () => {
		const result = render(<SimpleTest username={'@ansh'} age={22} email={'anshroshan@gmail.com'} />)
		expect(result.container).toMatchSnapshot()
	})
})
