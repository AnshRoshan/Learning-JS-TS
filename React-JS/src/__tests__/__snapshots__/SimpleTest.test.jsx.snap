// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Check if the Vitest is working > Snapshot testing : Simpletest 1`] = `
<div>
  <div>
    <h1>
      Simple Test
    </h1>
    <p>
      This is a simple test component
    </p>
    <div>
      <p>
        Username: 
        @ansh
      </p>
      <p>
        Email: 
        anshroshan@gmail.com
      </p>
      <p>
        Age: 
        22
      </p>
    </div>
  </div>
</div>
`;

exports[`Check if the Vitest is working > should render the component with text 1`] = `
<div>
  <div>
    <h1>
      Simple Test
    </h1>
    <p>
      This is a simple test component
    </p>
    <div>
      <p>
        Username: 
        @ansh
      </p>
      <p>
        Email: 
        anshroshan@gmail.com
      </p>
      <p>
        Age: 
        22
      </p>
    </div>
  </div>
</div>
`;
